lib/code_corps_web/controllers/organization_github_app_installation_controller.ex

```diff
@@ -22,9 +16,45 @@ defmodule CodeCorpsWeb.OrganizationGithubAppInstallationController do
     query |> id_filter(id_list)
   end
 
-  @spec handle_create(Plug.Conn.t, map) :: Ecto.Changeset.t
-  def handle_create(_conn, attributes) do
-    %OrganizationGithubAppInstallation{}
-    |> OrganizationGithubAppInstallation.create_changeset(attributes)
+  @spec index(Conn.t, map) :: Conn.t
+  def index(%Conn{} = conn, %{} = params) do
+    with organization_installations <- OrganizationGithubAppInstallation |> Query.id_filter(params) |> Repo.all do
+      conn |> render("index.json-api", data: organization_installations)
+    end
+  end
+
+  @spec show(Conn.t, map) :: Conn.t
+  def show(%Conn{} = conn, %{"id" => id}) do
+    with %OrganizationGithubAppInstallation{} = organization_installation <- OrganizationGithubAppInstallation |> Repo.get(id) do
+      conn |> render("show.json-api", data: organization_installation)
+    end
+  end
+
+  @spec create(Plug.Conn.t, map) :: Conn.t
+  def create(%Conn{} = conn, %{} = params) do
+    with %User{} = current_user <- conn |> Guardian.Plug.current_resource,
+         {:ok, :authorized} <- current_user |> Policy.authorize(:create, %OrganizationGithubAppInstallation{}, params),
+         {:ok, %OrganizationGithubAppInstallation{} = organization_installation} <- %OrganizationGithubAppInstallation{} |> OrganizationGithubAppInstallation.create_changeset(params) |> Repo.insert do
+      conn |> put_status(:created) |> render("show.json-api", data: organization_installation)
+    end
+  end
+
+  @spec update(Conn.t, map) :: Conn.t
+  def update(%Conn{} = conn, %{"id" => id} = params) do
+    with %OrganizationGithubAppInstallation{} = organization_installation <- OrganizationGithubAppInstallation |> Repo.get(id),
+      %User{} = current_user <- conn |> Guardian.Plug.current_resource,
+      {:ok, :authorized} <- current_user |> Policy.authorize(:update, organization_installation),
+      {:ok, %OrganizationGithubAppInstallation{} = organization_installation} <- organization_installation |> OrganizationGithubAppInstallation.changeset(params) |> Repo.update do
+        conn |> render("show.json-api", data: organization_installation)
+    end
+  end
+
+  @spec delete(Plug.Conn.t, map) :: Conn.t
+  def delete(%Conn{} = conn, %{} = params) do
+    with %User{} = current_user <- conn |> Guardian.Plug.current_resource,
+         {:ok, :authorized} <- current_user |> Policy.authorize(:delete, %OrganizationGithubAppInstallation{}, params),
+         {:ok, %OrganizationGithubAppInstallation{} = organization_installation} <- %OrganizationGithubAppInstallation{} |> OrganizationGithubAppInstallation.changeset(params) |> Repo.insert do
+      conn |> put_status(:deleted) |> render("show.json-api", data: organization_installation)
```

Comment:
The order of operations here should be

- load organization github app installation by id
- `{:ok, :authorized} <- current_user |> Policy.authorize(:delete, organization_github_app_installation)`,
- `{:ok, %OrganizationGithubAppInstallation{}} <- organization_github_app_installation |> Repo.delete`
- render a 204 (basically, `conn |> send_resp(:no_content, "")`

If you implement it that way, you do should 
- not have to make the changeset function public
- implement the delete policy to accept a record, not params

Reply:
Updated @begedin, thanks